extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/music/GoogleMusicExporter.java

```diff
@@ -0,0 +1,277 @@
+    Optional<String> paginationToken =
+        paginationData.map((PaginationData value) -> ((StringPaginationToken) value).getToken());
+
+    PlaylistItemListResponse playlistItemListResponse =
+        getOrCreateMusicHttpApi(authData).listPlaylistItems(playlistId, paginationToken);
+
+    PaginationData nextPageData = null;
+    if (!Strings.isNullOrEmpty(playlistItemListResponse.getNextPageToken())) {
+      nextPageData = new StringPaginationToken(playlistItemListResponse.getNextPageToken());
+    }
+    ContinuationData continuationData = new ContinuationData(nextPageData);
+
+    MusicContainerResource containerResource = null;
+    GooglePlaylistItem[] googlePlaylistItems = playlistItemListResponse.getPlaylistItems();
+    List<MusicPlaylistItem> playlistItems = new ArrayList<>();
+    if (googlePlaylistItems != null && googlePlaylistItems.length > 0) {
+      for (GooglePlaylistItem googlePlaylistItem : googlePlaylistItems) {
+        playlistItems.add(convertPlaylistItem(playlistId, googlePlaylistItem));
+      }
+      containerResource = new MusicContainerResource(null, playlistItems, null, null);
+    }
+
+    return new ExportResult<>(ResultType.CONTINUE, containerResource, continuationData);
+  }
+
+  private int getTokenPrefixLength(String token) {
+    final ImmutableList<String> knownPrefixes =
+        ImmutableList.of(
+            PLAYLIST_TRACK_RELEASE_TOKEN_PREFIX,
+            PLAYLIST_TRACK_TOKEN_PREFIX,
+            PLAYLIST_RELEASE_TOKEN_PREFIX,
+            PLAYLIST_TOKEN_PREFIX,
+            TRACK_RELEASE_TOKEN_PREFIX,
+            TRACK_TOKEN_PREFIX,
+            RELEASE_TOKEN_PREFIX);
+
+    for (String prefix : knownPrefixes) {
+      if (token.startsWith(prefix)) {
+        return prefix.length();
+      }
+    }
+    return 0;
+  }
+
+  private List<MusicGroup> createMusicGroups(String[] artistTitles) {
+    if (artistTitles == null) {
+      return null;
+    }
+    List<MusicGroup> musicGroups = new ArrayList<>();
+    for (String artistTitle : artistTitles) {
+      musicGroups.add(new MusicGroup(artistTitle));
+    }
+    return musicGroups;
+  }
+
+  private MusicPlaylistItem convertPlaylistItem(
+      String playlistId, GooglePlaylistItem googlePlaylistItem) {
+    GoogleTrack track = googlePlaylistItem.getTrack();
+    GoogleRelease release = track.getRelease();
+    return new MusicPlaylistItem(
+        new MusicRecording(
+            track.getIsrc(),
+            track.getTrackTitle(),
+            track.getDurationMillis(),
+            new MusicRelease(
+                release.getIcpn(),
+                release.getReleaseTitle(),
+                createMusicGroups(release.getArtistTitles())),
+            createMusicGroups(track.getArtistTitles())),
+        playlistId,
+        googlePlaylistItem.getOrder());
+  }
+
+  private synchronized GoogleMusicHttpApi getOrCreateMusicHttpApi(TokensAndUrlAuthData authData) {
+    return musicHttpApi == null ? makeMusicHttpApi(authData) : musicHttpApi;
```

Comment:
note for myself when I merge this later: need to fix `musicHttpApi` as it never gets set